docs/tutorials/mastering.md

```diff
@@ -0,0 +1,658 @@
+# Mastering
+
+In this tutorial, we will deploy and run the SD-Core 5G core network following Control and User Plane Separation (CUPS) principles.  The radio and cell phone simulator will also be deployed on an isolated cluster.
+
+```{image} ../images/sdcore_networking.png
+:alt: Network diagram
+:align: center
+```
+
+To complete this tutorial, you will need four machines (virtual or physical), with access to the networks as follows.
+
+| Machine | CPUs | RAM | Disk | Networks |
+| ------- | ---- | --- | ---- | -------- |
+| Control Plane Kubernetes Cluster     | 4 | 8g | 40g | `management` |
+| User Plane Kubernetes Cluster        | 4 | 4g | 20g | `management`, `access`, `core` |
+| Juju Controller + Kubernetes Cluster | 2 | 8g | 40g | `management` |
+| gNB Simulator Kubernetes  Cluster    | 4 | 4g | 20g | `management`, `ran` |
+
+
+## Sample Values
+
+For the purpose of this guide, the following values will be used.
+
+### Networks
+
+| Name | Subnet | Gateway IP |
+| ---- | ------ | ---------- |
+| `management` | 10.201.0.0/24 | 10.201.0.1 |
+| `access`     | 10.202.0.0/24 | 10.202.0.1 |
+| `core`       | 10.203.0.0/24 | 10.203.0.1 |
+| `ran`        | 10.204.0.0/24 | 10.204.0.1 |
+
+### IP Addresses
+
+The following IP addresses are used in this tutorial.
+
+| Name | IP Address | Purpose |
+| ---- | ---------- | ------- |
+| `juju-controller.mgmt` | 10.201.0.10 | Management address for Juju machine |
+| `control-plane.mgmt` | 10.201.0.11 | Management address for control plane cluster machine |
+| `user-plane.mgmt` | 10.201.0.12 | Management address for user plane cluster machine |
+| `gnbsim.mgmt` | 10.201.0.13 | Management address for the gNB Simulator cluster machine |
+| `api.juju-controller.mgmt` | 10.201.0.50 | Juju controller address |
+| `cos.mgmt` | 10.201.0.51 | Canonical Observability Stack address |
+| `amf.mgmt` | 10.201.0.100 | Externally reachable control plane endpoint for the AMF |
+| `nms.mgmt` | 10.201.0.101 | Externally reachable control plane endpoint for the NMS |
+| `upf.mgmt` | 10.201.0.200 | Externally reachable control plane endpoint for the UPF |
+
+## 1. Bootstrap all Machines
+
+This section covers how to install Juju and MicroK8s to act as the infrastructure for SD-Core, and may be skipped as needed if already done or a different Kubernetes deployment is being used.
+
+### SD-Core Control Plane
+
+All commands in this section are run on the SD-Core Control Plane Kubernetes cluster host.
+
+First, install Microk8s.
+
+```console
+sudo snap install microk8s --channel=1.27-strict/stable
+sudo microk8s enable hostpath-storage
+```
+
+The control plane needs to expose two services: the AMF and the NMS. In this next step, we enable the MetalLB add on in Microk8s, and give it a range of two IP addresses. The first IP address will be used by the AMF (`10.201.0.100`) and the second for the Network Management Service (`10.201.0.101`):
```

Comment:
Oh.  That made me realize that the AMF and NMS could potentially swap places depending on the order of charm initialization.  If NMS grabs `10.201.0.100` instead of the AMF, the tutorial will fail.

Reply:
We could rely on the fact that traefik takes about 30s longer to grab the IP
```
control-plane    amf-external                         LoadBalancer   10.152.183.73    10.201.0.100   38412:30431/SCTP                        4m18s
control-plane    traefik-k8s                          LoadBalancer   10.152.183.69    10.201.0.101   80:30001/TCP,443:30801/TCP              3m38s
```